DSpace/dspace-angular
# Fix minor collection form bugs

## References
* Fixes #2345
* Maintenance PR #2405

## Description
This PR hides the `Entity Type` field in both the create and edit collection form when the entities are not initialised yet. It also fixes a small bug in the collection, community, eperson & group form where hitting `ENTER` in a field would lead to other buttons from being triggered like cancel or go back which causes the form to close.

## Instructions for Reviewers
List of changes in this PR:
* `CollectionFormComponent`: hide the `Entity Type` field  when there are no entity types
* `ComColFormComponent`, `GroupFormComponent`, `EPersonFormComponent`: some buttons in the `<form>`s did not have the correct type which lead to incorrect actions being called when

**Guidance for how to test and review this PR:**
- Use an empty DB/one were entities are not initialised yet and check that the collection forms (create & edit) don't have the entity type field by default
- Run the script `initialize-entities -f ../dspace/config/entities/relationship-types.xml`and check that that field is now visible in both forms
- For the submission forms Community, Collection, EPerson & Group check that:
  - When hitting enter in an empty submission form you are not redirected to an incorrect page
  - When you hit enter in a correctly filled in form, that the `DSpaceObject` is correctly created

## Checklist
